hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -1,11 +1,72 @@
 module HookContextExtension
+  # FIXME: remove when #23332 is released
+  def param_value(mod, name)
+    param(mod, name).value if param(mod, name)
+  end
+
   def success_file
     File.join(File.dirname(kafo.config.config_file), '.installed')
   end
 
   def new_install?
     !File.exist?(success_file)
   end
+
+  def install_package(package)
+    execute("puppet resource package #{package} ensure=installed")
```

Comment:
Puppet may not be on `$PATH`. Calling `sudo foreman-installer` will not execute `/etc/profile.d/puppet-agent.sh`. For this Kafo has the ugly workaround that's [search_puppet_path](https://github.com/theforeman/kafo/blob/bbf9288473e0f1af9892c9cba0dc665be4ea51b3/lib/kafo/puppet_command.rb#L37-L42)

Reply:
So you are wanting me to copy that logic to here?